Framework/WorkflowAlgorithms/src/QENSFitSequential.cpp

```diff
@@ -0,0 +1,694 @@
+
+  const bool doExtractMembers = getProperty("ExtractMembers");
+  if (doExtractMembers)
+    extractMembers(groupWs, workspaces, outputBaseName + "_Members");
+
+  deleteTemporaryWorkspaces(outputBaseName);
+
+  addAdditionalLogs(resultWs);
+  copyLogs(resultWs, groupWs);
+
+  setProperty("OutputWorkspace", resultWs);
+  setProperty("OutputParameterWorkspace", outputWs);
+  setProperty("OutputWorkspaceGroup", groupWs);
+}
+
+std::map<std::string, std::string>
+QENSFitSequential::getAdditionalLogStrings() const {
+  bool convolve = getProperty("ConvolveMembers");
+  auto fitProgram = name();
+  fitProgram = fitProgram.substr(0, fitProgram.rfind("Sequential"));
+
+  auto logs = std::map<std::string, std::string>();
+  logs["sample_filename"] = getPropertyValue("InputWorkspace");
+  logs["convolve_members"] = convolve ? "true" : "false";
+  logs["fit_program"] = fitProgram;
+  logs["fit_mode"] = "Sequential";
+  return logs;
+}
+
+std::map<std::string, std::string>
+QENSFitSequential::getAdditionalLogNumbers() const {
+  std::map<std::string, std::string> logs;
+  logs["start_x"] = getPropertyValue("StartX");
+  logs["end_x"] = getPropertyValue("EndX");
+  return logs;
+}
+
+void QENSFitSequential::addAdditionalLogs(
+    MatrixWorkspace_sptr resultWorkspace) {
+  auto logAdder = createChildAlgorithm("AddSampleLog", -1.0, -1.0, false);
+  logAdder->setProperty("Workspace", resultWorkspace);
+
+  Progress logAdderProg(this, 0.99, 1.00, 6);
+  logAdder->setProperty("LogType", "String");
+  for (const auto log : getAdditionalLogStrings()) {
+    logAdder->setProperty("LogName", log.first);
+    logAdder->setProperty("LogText", log.second);
+    logAdder->executeAsChildAlg();
+    logAdderProg.report("Add text logs");
+  }
+
+  logAdder->setProperty("LogType", "Number");
+  for (const auto log : getAdditionalLogNumbers()) {
+    logAdder->setProperty("LogName", log.first);
+    logAdder->setProperty("LogText", log.second);
+    logAdder->executeAsChildAlg();
+    logAdderProg.report("Add number logs");
+  }
+}
+
+std::string QENSFitSequential::getOutputBaseName() const {
+  const auto base = getPropertyValue("OutputWorkspace");
+  auto position = base.rfind("_Result");
+  if (position != std::string::npos)
+    return base.substr(0, position);
+  return base;
+}
+
+bool QENSFitSequential::throwIfElasticQConversionFails() const { return false; }
+
+bool QENSFitSequential::isFitParameter(const std::string &) const {
+  return true;
+}
+
+std::vector<std::string> QENSFitSequential::getFitParameterNames() const {
+  IFunction_sptr function = getProperty("Function");
+
+  std::unordered_set<std::string> nameSet;
+  std::vector<std::string> names;
+  names.reserve(function->nParams());
+
+  for (auto i = 0u; i < function->nParams(); ++i) {
+    auto name = shortParameterName(function->parameterName(i));
+
+    if (isFitParameter(name) && nameSet.find(name) == nameSet.end()) {
+      names.emplace_back(name);
+      nameSet.insert(name);
```

Comment:
This looks awfully complicated. Why not just use and return `std::set`? No need to explicitly check if a names is there already. Also avoids cloning the names into two data structures.

Reply:
I agree this is complicated - the reason this was done is because the fit parameters need to be placed uniquely into a vector as this is then used to set a property value ('ParameterNames' property of ProcessIndirectFitParameters algorithm).

Reply:
How about this?
```c++
{
    IFunction_sptr function = getProperty("Function");
    std::set<std::string> nameSet;
    for (auto i = 0u; i < functions->nParams(); ++i) {
        const auto name = shortParameterName(function->parameterName(i));
        nameSet.insert(name);
    }
    return std::vector<std::string>(nameSet.cbegin(), nameSet.cend()):
}
```